homework/index.js

```diff
@@ -30,17 +34,110 @@
+      repositories.forEach(repo => {
+        if (repo.name === repoName) {
+          const li = createAndAppend('li', ul);
+          const table = createAndAppend('table', li);
+          const tbody = createAndAppend('tbody', table);
+          const tr1 = createAndAppend('tr', tbody);
+          createAndAppend('td', tr1, { text: `Name:`, class: 'label' });
+          const a = createAndAppend('a', tr1, { href: `${repo.html_url}` });
+          createAndAppend('td', a, { text: `${repo.name}` });
+          addRow(tbody, 'Description', `${repo.description}`);
+          addRow(tbody, 'Forks', `${repo.forks}`);
+          addRow(tbody, 'Updated', `${repo.updated_at.slice(0, 10)}`);
+        }
+      });
+    });
+  }
+
+  function renderContributions(repoName, ul) {
+    fetchJSON(
+      `https://api.github.com/repos/HackYourFuture/${repoName}/contributors`,
+      (err, repositories) => {
+        if (err) {
+          renderError(err);
+          return;
+        }
+        ul.innerHTML = '';
+        createAndAppend('h3', ul, { text: `Contributors:` });
+        repositories.forEach(repo => {
+          const a = createAndAppend('a', ul, { href: `${repo.html_url}` });
+          const li = createAndAppend('li', a);
+          const table = createAndAppend('table', li);
+          const tbody = createAndAppend('tbody', table);
+          const tr1 = createAndAppend('tr', tbody);
+          createAndAppend('img', tr1, { src: `${repo.avatar_url}` });
+          createAndAppend('td', tr1, { text: `${repo.login}` });
+          createAndAppend('td', tr1, { text: `${repo.contributions}` });
+        });
+      },
+    );
+  }
+
+  function createOptionElements(repositories, select) {
+    createAndAppend('option', select, { text: 'Select a repository', disabled: `disabled` });
+    repositories
+      .sort((a, b) => a.name.localeCompare(b.name))
+      .forEach(repo => {
+        createAndAppend('option', select, { text: repo.name, value: repo.name });
+      });
+  }
+
   function main(url) {
+    const root = document.getElementById('root');
+    const body = document.body;
+    const header = document.createElement('header');
+    body.prepend(header);
+    createAndAppend('h1', header, { text: 'HYF Repositories' });
+    const select = createAndAppend('select', header);
+
+    const ul = createAndAppend('ul', root, { id: 'list-container' });
+    const ul1 = createAndAppend('ul', root, { id: 'list-contributions' });
+
     fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+
+      function setDefault() {
+        const repoName = 'alumni';
+        renderRepos(repoName, ul);
+        renderContributions(repoName, ul1);
+      }
+      setDefault();
```

Comment:
The benefit of this short directly invoked function is questionable. You could simply reduce it to using lines 125-126 directly. Taking note of my other comments, to display the first repo you could simply do:

```js
renderRepos(repositories[0], ul);
renderContributions(repositories[0], ul1);
```

Reply:
You gave me a thumbs up but didn't change your code 😞.

Reply:
`function setDefault() {
        renderRepos(repositories[0], tableContainer);
        renderContributions(repositories[0], ul);
      }
      setDefault();`
Isn't this the right way?

Reply:
I mean you do not need this immediately invoked function at all. It doesn't add value. You can just use the code directly in-line:

```js
...
createOptionElements(repositories, select);

renderRepos(repositories[0], tableContainer);
renderContributions(repositories[0], ul);

select.addEventListener('change', () => {
  const repo = repositories[select.value];
  renderRepos(repo, tableContainer);
  renderContributions(repo, ul);
});
...
```